gtsfm/frontend/correspondence_generator/keypoint_aggregator/keypoint_aggregator_base.py

```diff
@@ -0,0 +1,45 @@
+"""Base class for keypoint aggregators.
+
+Authors: John Lambert
+"""
+
+import abc
+from typing import Dict, List, Optional, Tuple
+
+import dask
+from dask.delayed import Delayed
+
+from gtsfm.common.keypoints import Keypoints
+
+
+class KeypointAggregatorBase:
+    """Base class for keypoint aggregators."""
+
+    @abc.abstractmethod
+    def run(self, keypoints_dict: Dict[Tuple[int, int], Tuple[Keypoints, Keypoints]]) -> List[Optional[Keypoints]]:
+        """Aggregates per-pair image keypoints into a set of keypoints per image, without de-duplication.
```

Comment:
> without de-duplication
This is not guaranteed, right?

Reply:
Good catch, this comment only applies to `KeypointAggregatorUnique`, not `KeypointAggregatorDedup`. Updated.